app/controllers/topbar_sweeper.rb

```diff
@@ -18,7 +18,8 @@ def after_select(record)
   end
 
   def self.fragment_name(id = User.current.id)
-    "tabs_and_title_records-#{id}"
+    @@top_menu_cache_hash ||= Menu::Manager.items(:top_menu).content_hash
+    "tabs_and_title_records-#{@@top_menu_cache_hash}-#{id}"
   end
```

Comment:
FYI this stores the hash in a class member so it won't change anymore, but the only way of adding/removing a menu item at the moment is via plugin (un)installation, therefore this works fine. We do not want to calculate the hash on each request.